features/step_definitions/assertion_steps.rb

```diff
@@ -21,3 +21,12 @@
 Then("I should be on landing page") do
   expect(current_path).to eq root_path
 end
+
+Then("I should be on offer page") do
+  expect(current_path).to eq new_listing_offer_path
+end
+
+Then("I should be on the {string} listing page") do |listing_name|
+  name = Listing.find_by(pet_name: listing_name)
+  visit listing_path(name)
```

Comment:
This is an assertion step, you should test that you are on a certain page. Here you are telling the application to go to a specific page

Reply:
I misunderstood, changed the assertion step to expect instead